### DISCUSSION TITLE
How does the squash function work

### DISCUSSION THREAD

**Discussion by vin-d** [Murmele/Gittyup#434] [Q&A]
I assume a squash would work that way : 
1) multiselect multiple commits by using shift+click 
2) use squash command (ctrl+shift+q)
3) the history get rewritten to remove the commits that are selected and part of the current branch
4) a new commit is done on HEAD which is the sum of all the selected commits.

from the log panel a squash does a merge command and I cannot get it to do anything.
Is it because not yet implemented ? (I see no issue related to squash)

**Comment by Murmele** [comment]
Hi @vin-d can you write an issue I will check it out

**Reply by PastorPL** [reply]
I will also would like to know more about squashing commits. I can see that there is such an option but how to use it? Use Case: I've got 3 local commits that I would like to squash before pushing to remote - how to do that?

**Reply by KaKi87** [reply]
Hello, any news on this ? Thanks

**Comment by Murmele** [comment]
Hmm it seems that squash is not correctly implemented. My expectation: is to get one commit from the current commit and the selected commit.

Original:
![image](https://user-images.githubusercontent.com/10099533/224825263-47efe3b2-30f5-4955-bd57-5ac022b850e8.png)

If I select a917d6b and do a squash I expect a single commit of f0173942c552f1606fc75b21e14c54b7dadc5ccc, 304041c7d0ed8d43f04bd6f5b6cdca4f2b26ec2b and 9d982fb42ef1ac3a4f350fa25ceed2918f5e8aee

**Comment by safetycar** [comment]
I wonder, if this is still not implemented, if it would be possible to implement it in a way that you could ctrl+click commits and squash only the selected commits, because I wouldn't have expected it to work as a range.
